tests/test_tool_output_truncation.py

```
@@ -106,3 +111,29 @@ def test_action_result_to_tool_result_preserves_output_metadata() -> None:
+        policy=ToolExecutionPolicy(
+            max_output_chars=160,
+            artifact_dir=tmp_path,
+            truncation_direction="tail",
+        ),
+    )
+
+    assert result.truncated is True
+    assert len(result.output) <= 160
+    assert "Tool output truncated" in result.output
+    assert result.artifact_path is not None
+    assert Path(result.artifact_path).read_text(encoding="utf-8") == "line\n" * 100
```

Ruff (0.15.12)

[warning] Async functions should not use pathlib.Path methods, use trio.Path or anyio.path

(ASYNC240)